pipcompilemulti/features/controller.py

```diff
@@ -61,9 +64,10 @@ def save_command_options(*args, **kwargs):
 
     def pin_options(self, env_name):
         """Return list of options to pin command."""
-        options = self.use_cache.pin_options()
+        options = self.use_cache.pin_options().copy()
```

Comment:
@peterdemin a copy is necessary here otherwise we're extending (and thus polluting) the class `enabled_pin_options` or `disabled_pin_options` lists.

Reply:
Agree